test/attr/attr_completionhandlerasync.swift

```diff
@@ -137,3 +142,21 @@ func matchingAsyncFunc(value: String) async {} // expected-note{{'matchingAsyncF
 // expected-error@+1:25{{ambiguous '@completionHandlerAsync' async function 'matchingAsyncFunc(value:)'}}
 @completionHandlerAsync("matchingAsyncFunc(value:)")
 func typecheckFunc9(handler: @escaping () -> Void) {}
+
+// Suggest using async alternative function in async context
+
```

Comment:
Might be worth adding some tests for the closure case above. Another case is to check we don't output a warning for references of either top level/instance functions.